### DISCUSSION TITLE
Handle unexpected value in request

### DISCUSSION THREAD

**Discussion by sagarnayak** [rwf2/Rocket#1757] [Questions] [answered]
If i have this route.

	#[get("/hello/<name>/<age>/<cool>")]
	fn hello(name: &str, age: u8, cool: bool) -> String {
		if cool {
			format!("You're a cool {} year old, {}!", age, name)
		} else {
			format!("{}, we need to talk about your coolness.", name)
		}
	}

and I provide an age number which is a number out of range from u8 then how do I handle this and send a meaningful message to the user.

As of now, I am thinking to create another route that takes a string age and tries to find the error that the user has one and provide a message.

**Comment by SergioBenitez** [comment, marked as answer]
Use a `Result<u8, &str>` instead of the bare `u8`. It will be an `Err(&str)` with the original parameter if and when it failed
to parse as a `u8`. All of this is discussed in detail in the [API docs for `FromParam`](https://api.rocket.rs/v0.5-rc/rocket/request/trait.FromParam.html#catching-errors).
